hash.c

```diff
@@ -6528,68 +6539,235 @@ env_update(VALUE env, VALUE hash)
 }
 
 /*
- *  A Hash is a dictionary-like collection of unique keys and their values.
- *  Also called associative arrays, they are similar to Arrays, but where an
- *  Array uses integers as its index, a Hash allows you to use any object
- *  type.
+ *  A \Hash is a dictionary-like collection of key-value pairs
+ *  whose keys are unique. (The values need not be unique.)
+ *
+ *  A \Hash has certain similarities to an \Array, but:
+ *  - An \Array index is always an \Integer.
+ *  - A \Hash key can be (almost) any object.
  *
- *  Hashes enumerate their values in the order that the corresponding keys
- *  were inserted.
+ *  === \Hash \Data Syntax
  *
- *  A Hash can be easily created by using its implicit form:
+ *  The older syntax for \Hash data uses the "hash rocket," <tt>=></tt>
+ *  (known in some languages as the <i>fat comma.</i>):
  *
- *    grades = { "Jane Doe" => 10, "Jim Doe" => 6 }
```

Comment:
We lost some context. New users trying to understand it might appreciate a slightly more tangible example.

Reply:
This one is straight out of the old doc (just relocated).  I don't want to mess with it.  Suggestion?

Reply:
Unchanged.  Not my code.